packages/react/src/components/CommandCenter/create.tsx

```diff
@@ -0,0 +1,146 @@
+import { observer } from "mobx-react-lite";
+import { CommandCenter as CommandCenterComponent } from "./CommandCenter";
+import { SuggestionsStore } from "./SuggestionsStore";
+import { CommandCenterSuggestions as CommandCenterSuggestionsComponent } from "./Suggestions";
+import { useEffect, useRef } from "react";
+import { Input } from "./Input";
+import { useKeybinding } from "@/hooks/keyboardEventHandler";
+import { KEY_BINDINGS } from "@/constants/keybindings";
+import { Suggestion } from "@/constants/commandTypes";
+import { ProjectContextType, useProjectContext } from "@/contexts/Project/useProjectContext";
+import { InputStore } from "./InputStore";
+import { CommandCenterStore } from "./CommandCenterStore";
+
+
+const createCommandCenterInput = ({ suggestionsStore, inputStore, onSuggestionSelect }: {
+    suggestionsStore: SuggestionsStore,
+    inputStore: InputStore,
+    onSuggestionSelect: (suggestion?: Suggestion) => void
+}) => {
+    const handleInputChange = (value: string) => {
+        inputStore.value = value;
+    }
+
+    const CommandCenterInput = observer(() => {
+        const inputRef = useRef<HTMLInputElement>(null);
+
+        useKeybinding({
+            action: () => {
+                onSuggestionSelect(suggestionsStore.highlightedSuggestion);
+            },
+            keys: KEY_BINDINGS.ENTER,
+            element: inputRef
+        });
+
+        useKeybinding({
+            action: () => suggestionsStore.highlightNextSuggestion(),
+            keys: KEY_BINDINGS.ARROW_DOWN,
+            element: inputRef
+        });
+
+        useKeybinding({
+            action: () => suggestionsStore.highlightPreviousSuggestion(),
+            keys: KEY_BINDINGS.ARROW_UP,
+            element: inputRef
+        });
```

Comment:
The input here is responsible for holding on to the focus and thus gets the keyboard events added to it.